scipy/optimize/nonlin.py

```diff
@@ -217,8 +218,7 @@ def _safe_norm(v):
 
 
 def _set_doc(obj):
-    if obj.__doc__:
-        obj.__doc__ = obj.__doc__ % _doc_parts
```

Comment:
This seems like it will break when `obj.__doc__ is None`, is this change necessary?

Reply:
(Also this is likely the source of the CircleCI failure)